torch/csrc/jit/codegen/cuda/compute_at.cpp

```diff
@@ -249,6 +239,29 @@ unsigned int ComputeAt::forwardComputeAt_impl(
     unsigned int producer_compute_at_axis) {
   FUSER_PERF_SCOPE("forwardComputeAt_impl");
 
+  // Can get into a situation where we inlined into a reduction, but then would
+  // try to traverse forward at that position but wouldn't be valid... e.g.:
+  // T1[ iS2{9}, iS20{( ceilDiv(5, 4) )}, iS21{4} ] compute_at( 3 )
+  //   -> T3[ iS6{9}, iS18{( ceilDiv(5, 4) )}, iS19{4} ] compute_at( 3 ) 3
+  // T3[ iS6{9}, iS18{( ceilDiv(5, 4) )}, iS19{4} ] compute_at( 3 )
+  //   -> T5[ iS12{9}, iS14{( ceilDiv(5, 4) )}rf, rS15{4}rf ] 3
+  // T5[ iS12{9}, iS14{( ceilDiv(5, 4) )}rf, rS15{4}rf ]
+  //   -> T4[ iS16{9}, rS17{( ceilDiv(5, 4) )} ] 3
+  // Reduce position to be inside first reduction
+  unsigned int first_red_pos = producer->nDims();
```

Comment:
@naoyam any idea why this wasn't an issue before? Seems like we would have done this previously, but just got an error in this PR during lowering complaining about this. Seems reasonable to move producer position here to be within first reduction as otherwise it seems clear like it would make an error. Had trouble detangling what I changed to have exposed this.